app/src/main/java/com/kamron/pogoiv/MainActivity.java

```diff
@@ -717,18 +718,21 @@ private void scanPokemon(Bitmap pokemonImage, String filePath) {
         //WARNING: this method *must* always send an intent at the end, no matter what, to avoid the application hanging.
         Intent info = Pokefly.createNoInfoIntent();
 
-        estimatedPokemonLevel = trainerLevel + 1.5;
-        estimatedPokemonLevel = getPokemonLevel(pokemonImage);
-
-        String pokemonName = getPokemonNameFromImg(pokemonImage);
-        String candyName = getCandyNameFromImg(pokemonImage);
-        int pokemonHP = getPokemonHPFromImg(pokemonImage);
-        int pokemonCP = getPokemonCPFromImg(pokemonImage);
+        try {
+            double estimatedPokemonLevel = getPokemonLevelFromImg(pokemonImage, trainerLevel);
+            String pokemonName = getPokemonNameFromImg(pokemonImage);
+            String candyName = getCandyNameFromImg(pokemonImage);
+            int pokemonHP = getPokemonHPFromImg(pokemonImage);
+            int pokemonCP = getPokemonCPFromImg(pokemonImage);
+            if (candyName.equals("") && pokemonHP == 10 && pokemonCP == 10){ //the default values for a failed scan, if all three fail, then probably scrolled down.
+                Toast.makeText(MainActivity.this, "Please scroll up and scan again", Toast.LENGTH_SHORT).show();
```

Comment:
Let's use a resource string so people can translate it. I added scan_pokemon_failed before:
https://github.com/farkam135/GoIV/pull/168/commits/4ee1aa7f031357bbdea8626b6bc8c216419b7be9
but your string might be better.